src/Tests/MiddlewareTest.cpp

```diff
@@ -60,16 +61,21 @@ class PluginTest
                 middleware[i] = new MiddleWare(std::move(*synth), &config);
                 master[i] = middleware[i]->spawnMaster();
                 //printf("Octave size = %d\n", master[i]->microtonal.getoctavesize());
+                if (i != NUM_MIDDLEWARE-1) {
+                    delete synth;
```

Comment:
This is required even with the std::move? I tend to forget some of those semantics.

Reply:
[Moving an object does not change its lifetime](https://stackoverflow.com/questions/15663539/why-isnt-object-returned-by-stdmove-destroyed-immediately). Compare it with an `std::vector` that you have on the stack - the move only alters its pointers. After you move from it, the memory is still there. Same with a vector (or `SYNTH_T`) on heap.

Aside from the theory, I could see with valgrind that this fixed a leak.

Reply:
Works for me, merging now.